test/inheritAttributes.test.js

```diff
@@ -84,3 +324,163 @@ QUnit.test('end to end', function(assert) {
+          <Representation
+            bandwidth="5000000"
+            codecs="avc1.64001e"
+            height="404"
+            id="test"
+            width="720">
+          </Representation>
+        </AdaptationSet>
+        <AdaptationSet mimeType="text/vtt" lang="en">
+          <Representation bandwidth="256" id="en">
+            <BaseURL>https://example.com/en.vtt</BaseURL>
+          </Representation>
+        </AdaptationSet>
+      </Period>
+    </MPD>
+  `
+  ));
+
+  const expected = [{
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.example.com/base/segments/',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: {},
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.example.com/base/media/',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: {},
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.test.com/base/segments/',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: {},
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.test.com/base/media/',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: {},
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '256',
+      baseUrl: 'https://example.com/en.vtt',
+      id: 'en',
+      lang: 'en',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'text/vtt',
+      periodIndex: 0,
+      role: {},
+      sourceDuration: 30
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }, {
+    attributes: {
```

Comment:
The subtitle representation gets duplicated here because of the way alternate base urls are generated. There are 2 parent level base urls, so we resolve this base url wrt both of those, however since this base url is absolute, we resolve the absolute url twice. 

This shouldn't really be an issue because in [organizeVttPlaylists](https://github.com/videojs/mpd-parser/blob/master/src/toM3u8.js#L75) we skip vtt playlists that we already have a playlist for that language. 

If we want to not produce these duplicates, then the base url creation logic will have to account for absolute urls lower down the tree and filter out duplicates. Thoughts?

Reply:
We may have more complicated vtt playlist support in the future that might involve different tracks with the same language and type (maybe). Perhaps dedupe in `organizeVttPlaylists` if we know it will always be unsegmented?

Reply:
Actually let's leave that as a TODO and get to it when we need to

Reply:
Ya we will probably want to dedupe based on AdaptionSet ID since that seems to be where the lang is defined